src/schemes/solid/discrete_element_method/contact_models.jl

```diff
@@ -0,0 +1,174 @@
+
+# Fields
+- `elastic_modulus::Float64`: Material Young's modulus (``E``).
+- `poissons_ratio::Float64`: Material Poisson's ratio (``\nu``).
+"""
+struct HertzContactModel <: ContactModel
+    elastic_modulus::Float64  # Material elastic modulus
+    poissons_ratio::Float64   # Material Poisson's ratio
+end
+
+@inline function collision_force_normal(model::HertzContactModel,
+                                        particle_system, neighbor_system,
+                                        overlap, normal, v_particle_system,
+                                        v_neighbor_system,
+                                        particle, neighbor, damping_coefficient)
+    # Use material properties from the Hertz contact model.
+    E_a = model.elastic_modulus
+    nu_a = model.poissons_ratio
+
+    # TODO: needs to be dispatched and handled for boundaries
+    # For the neighbor, use its properties if available (otherwise, assume identical material).
+    if neighbor_system isa DEMSystem && neighbor_system.contact_model isa HertzContactModel
+        E_b = neighbor_system.contact_model.elastic_modulus
+        nu_b = neighbor_system.contact_model.poissons_ratio
+    else
+        E_b = E_a
+        nu_b = nu_a
+    end
+
+    # Compute relative velocity along the contact normal.
+    v_a = current_velocity(v_particle_system, particle_system, particle)
+    v_b = current_velocity(v_neighbor_system, neighbor_system, neighbor)
+    rel_vel = v_a - v_b
+    rel_vel_norm = dot(rel_vel, normal)
+
+    # Compute the effective elastic modulus (Hertzian theory).
+    E_star = 1 / (((1 - nu_a^2) / E_a) + ((1 - nu_b^2) / E_b))
+
+    # Compute the effective radius.
+    r_a = particle_system.radius[particle]
+    r_b = neighbor_system.radius[neighbor]
+    r_star = (r_a * r_b) / (r_a + r_b)
+
+    # Non-linear stiffness for Hertzian contact.
+    elastic_force_per_overlap = (4 / 3) * E_star * sqrt(r_star * overlap)
+
+    # Compute effective mass for damping.
+    if neighbor_system isa DEMSystem
+        m_a = particle_system.mass[particle]
+        m_b = neighbor_system.mass[neighbor]
+        m_star = (m_a * m_b) / (m_a + m_b)
+    else
+        m_star = particle_system.mass[particle]
+    end
+
+    # Critical damping coefficient.
+    gamma_c = 2 * sqrt(m_star * elastic_force_per_overlap)
+
+    # Total normal force: elastic + damping term.
+    force_magnitude = elastic_force_per_overlap * overlap +
+                      damping_coefficient * gamma_c * rel_vel_norm
+
+    return force_magnitude * normal
+end
+
+@doc raw"""
+    LinearContactModel(; normal_stiffness)
+
+Linear spring-dashpot contact model ([Cundall1979](@cite)).
+
+This model calculates the normal contact force between two objects based on a linear spring
+law for the elastic component and a linear viscous damping law for the dissipative component.
+The total normal force ``F_n`` is given by:
+```math
+F_n = k_n \delta + \gamma_d v_{\text{rel,n}}
```

Comment:
```suggestion
The total normal force ``F_n`` is given by
```math
F_n = k_n \delta + \gamma_d v_{\text{rel,n}},
```